### PR TITLE
Add new NSS module

### PR DESCRIPTION
**Pull request by hopkiw**
Adding just the parsing functions and tests, to start. This is in a separate file that will not be built and packaged, though the tests will be run as part of presubmits.

### CODE REVIEW

**Comment by illfelder** [comment]
/cc @sarsate

Thanks for breaking out this logic into a separate PR. I'm adding Danny who I think will do a much better job catching issues.

**Review by Sarsate — APPROVED** [review]
LGTM!

**Review by winterfreshx — COMMENTED** [review]
I had some questions about the function and the use of buffers. I am not sure how the lifecycle of the buffer and the struct will be handled. There is an implicit connection between the single string buffer that is reused for storing all the string values of the structs. Is the default buffer going to be taken from the heap as it is in the test?

File: src/nss/new_nss_oslogin.c [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  if (fields[PW_END] > (int)buflen) {
+    return ERANGE;
+  }
+
+  int i, field;
+  for(field = 1, i = 0; i < fields[PW_END]; i++) {
+    if (str[i] == ':') {
+      fields[field++] = i+1;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (hopkiw) [reply src/nss/new_nss_oslogin.c:82]: `strtok` would work here except for two things: first, we will end up doing work and may only find the string is malformed or too large at the end. scanning first for how many colon-separated fields there are allows an early return. secondly, `strtok` doesn't support sequential delimiters, which can occur in passwd/group fields (for example, if GECOS is omitted), or delimiters at start or end of string

**Comment by winterfreshx** [comment]
LGTM

**Pull request merged**